client/src/components/Menu/index.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import withStyles from 'isomorphic-style-loader/withStyles';
+import Hamburger from 'react-hamburger-menu';
+import MenuItem from './MenuItem';
+
+import s from './Menu.css';
+
+class Menu extends React.Component {
+  static propTypes = {
+    currentTab: PropTypes.string.isRequired,
+  };
+
+  // eslint-disable-next-line react/state-in-constructor
+  state = {
+    isMenuOpen: false,
+  };
+
+  onMenuClick = () => {
+    const { isMenuOpen } = this.state;
+    this.setState({ isMenuOpen: !isMenuOpen });
+  };
+
+  render() {
+    const { currentTab } = this.props;
+    const { isMenuOpen } = this.state;
+    return (
+      <div className={s.menu}>
+        <div className={s.menuLogo}>IoT App</div>
+        <div className={s.menuHamburger}>
+          <Hamburger
+            isOpen={isMenuOpen}
+            menuClicked={this.onMenuClick}
+            width={25}
+            height={15}
+            color="#3f496f"
+          />
+        </div>
+        <div className={`${s.menuItems} ${isMenuOpen && s.menuItemsActive}`}>
+          {this.menuItems.map(item => (
+            <MenuItem
+              key={item.text}
+              item={item}
+              isActive={currentTab === item.text}
+              closeMenu={this.onMenuClick}
+            />
+          ))}
+        </div>
+        <div className={s.menuSocial}>
+          <a
+            href="https://google.com"
+            target="_blank"
+            rel="noreferrer noopener"
+            className={s.menuSocialIcon}
+          >
+            <svg
+              width="18px"
+              height="18px"
+              fill="rgba(255, 255, 255, .6)"
+              viewBox="0 0 496 512"
+            >
+              <path d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm193.2 152h-82.5c-9-44.4-24.1-82.2-43.2-109.1 55 18.2 100.2 57.9 125.7 109.1zM336 256c0 22.9-1.6 44.2-4.3 64H164.3c-2.7-19.8-4.3-41.1-4.3-64s1.6-44.2 4.3-64h167.4c2.7 19.8 4.3 41.1 4.3 64zM248 40c26.9 0 61.4 44.1 78.1 120H169.9C186.6 84.1 221.1 40 248 40zm-67.5 10.9c-19 26.8-34.2 64.6-43.2 109.1H54.8c25.5-51.2 70.7-90.9 125.7-109.1zM32 256c0-22.3 3.4-43.8 9.7-64h90.5c-2.6 20.5-4.2 41.8-4.2 64s1.5 43.5 4.2 64H41.7c-6.3-20.2-9.7-41.7-9.7-64zm22.8 96h82.5c9 44.4 24.1 82.2 43.2 109.1-55-18.2-100.2-57.9-125.7-109.1zM248 472c-26.9 0-61.4-44.1-78.1-120h156.2c-16.7 75.9-51.2 120-78.1 120zm67.5-10.9c19-26.8 34.2-64.6 43.2-109.1h82.5c-25.5 51.2-70.7 90.9-125.7 109.1zM363.8 320c2.6-20.5 4.2-41.8 4.2-64s-1.5-43.5-4.2-64h90.5c6.3 20.2 9.7 41.7 9.7 64s-3.4 43.8-9.7 64h-90.5z" />
```

Comment:
this look ugly, maybe extract it as a separate component?

Reply:
but even if we extract it in separate component we still have this array of svgs, i hardcode it cause its faster than import that takes some time to render

Reply:
Code prettiness is more important than execution performance by default.